### DISCUSSION TITLE
Traditional Animation or Keyframe Animation

### DISCUSSION THREAD

**Discussion by keiwando** [keiwando/toonsquid#3] [Polls]
ToonSquid allows you to work both in the style of traditional frame-by-frame animation and also using keyframes. If you animate with ToonSquid, I would love to know which one of these two techniques you use most often.

**Comment by emmettcrass** [comment]
I mix them. Some things need frame-by-frame but others can get away with keyframes.

**Comment by Scvosh** [comment]
Аналогичная ситуация! Имеет место быть и тому и тому, но руку приложить стоит к keyframes. Так как она даёт больше возможностей хотябы для тех кто не столь хорош в покадровой анимации

**Comment by kevincreative** [comment]
While I mainly use keyframing, having the frame by frame ability is great to have when you need it.

**Comment by DWrightAnimation** [comment]
I usual use Traditional workflow, but it would be nice to have more advanced rigging options, such as Z-Axis nudging, Inverse Kinematics, a keyframe library to reuse positions, rotations, scaling, etc., because right now, ToonSquid's rigging tools are the most convenient option for me to make storytime animations for YouTube.

**Comment by kevincreative** [comment]
Yes IK, FK, bones, etc as well as some sort of distortion or warping transforms would be awesome to have. For now though the system is not too bad and is super easy to use for 2d flat style simple rigging.

**Comment by amoraleite** [comment]
Guys, Is there any way to change between the two styles? Or when I choose one do I have to stick with the chosen model?

**Reply by Cat-Butler** [reply]
You can happily mix and change between the two styles in ToonSquid, either in different layers or within the same (timeline) layer.

**Reply by amoraleite** [reply]
thanks ! but I'm not able to do this. When I start in keyframe mode and try to add a frame it adds an image that extends on the timeline and I have to reduce it manually all the time. And when I start in traditional mode I lose the hierarchy ability. what am I wrong?

**Reply by keiwando** [reply]
Here's more info on the default workflow choice that you can make when you create project:
https://toonsquid.com/handbook/projects/create_project/#workflow
You can change the default duration of new drawings later in the settings and you can also always switch the transform tool mode between editing pixels within a layer or editing the transform of a pixel layer:
https://toonsquid.com/handbook/transform/transform_tool/#edit-pixels
